lib/procore/requestable.rb

```diff
@@ -364,7 +362,11 @@ def multipart?(body)
     end
 
     def full_path(path)
-      File.join(base_api_path, path).to_s
+      if /(vapid|rest)/ =~ path
+        File.join(base_api_path, path)
+      else
+        File.join(base_api_default_path, path)
+      end.to_s
```

Comment:
Okay, I'm happy with this as a way to at least preserve backward compatibility. The regex needs some fixing however - `/(vapid|rest)/ =~ 'something-with-rest-anywhere-in-string' => true`.

Regex should be something like:

```ruby
# \A beginning of string
# \/? One or 0 forward slashes
# (rest|vapid) the string 'rest' or 'vapid'
regex = /\A\/?(rest|vapid)/
```

Reply:
Actually, now that I think about it, I'm going to go ahead and say that the confusion this behavior is going to cause is going to be greater than just removing some behavior. A few scenarios:

```ruby
# Using SDK to communicate with vapid
# Prior to upgrade:
client.get('me')
=> /vapid/me

# After upgrade (going into the `else` case)
=> /rest/1.0/me

# Using with 1.0. In this case it works okay, but confusing if a resource is on a minor bump, because why do 
# I have to specify a version in one call, but not the other?
client.get('me')
=> /rest/1.0/me

client.get('/rest/1.1/updated-resource)
=> /rest/1.1/updated-resource

# V2 is released, this gem is updated, but someone is running an old version of the gem
client.get('me')
=> /rest/1.0/me
```

So I would prefer we just force the full path usage, and not try and guess what the user actually wants. If the app want to only use vapid, they can update their client initializer and set the base_api_path to `app.procore.com/vapid` and keep the old behavior. So I would back out this change and keep it as 

```ruby
def full_path(path)
  File.join(base_api_path, path).to_s
end
```

Reply:
I agree the switch can feel intuitive, but I think having a default provides some benefits.

With the new default, the migration is completely painless. It's just upgrading the gem. If we enforce a full path they'll need to change all instances of `client.get('me')` to `client.get('rest/v1.0/me')`. They could use `https://api.procore.com/rest/v1.0` as the host, but then can no longer choose versions for different endpoints. 

We want to encourage developers to use the `v1.0` version for most of their requests, and only the minor versions `v.1.1`, `v1.2`, etc. when they have a need for them. It makes sense why `/rest/1.1` requires a version number in the path but `/rest/1.0` does not. The average integrator can ignore API versioning unless it applies to them, and use the SDK exactly as they had before.

Reply:
@PatKoperwas Made some major changes to the interface. Take a look!